packages/backend/src/plugins/DemoEventBasedEntityProvider.ts

```diff
@@ -18,42 +18,40 @@ import {
   EntityProvider,
   EntityProviderConnection,
 } from '@backstage/plugin-catalog-node';
-import {
-  EventBroker,
-  EventParams,
-  EventSubscriber,
-} from '@backstage/plugin-events-node';
+import { EventParams, EventService } from '@backstage/plugin-events-node';
 import { Logger } from 'winston';
 
-export class DemoEventBasedEntityProvider
-  implements EntityProvider, EventSubscriber
-{
+export class DemoEventBasedEntityProvider implements EntityProvider {
   private readonly logger: Logger;
+  private readonly eventService: EventService;
   private readonly topics: string[];
 
   constructor(opts: {
-    eventBroker: EventBroker;
+    eventService: EventService;
     logger: Logger;
     topics: string[];
   }) {
-    const { eventBroker, logger, topics } = opts;
-    this.logger = logger;
-    this.topics = topics;
-    eventBroker.subscribe(this);
+    this.eventService = opts.eventService;
+    this.logger = opts.logger;
+    this.topics = opts.topics;
+  }
+
+  async subscribe() {
+    await this.eventService.subscribe({
+      id: this.constructor.name,
+      topics: this.topics,
+      onEvent: this.onEvent.bind(this),
```

Comment:
Another option is to declare `onEvent` as an arrow function: `private onEvent = async (...) => { ... }`

Reply:
Yes, you are right. I just reused the existing method. However, I could replace it with the arrow function instead. Would you prefer that? Maybe better for the `DemoEventBasedEntityProvider` in particular. 🤔